### PR TITLE
Sort by new + pages

### PR DESCRIPTION
**Pull request by gabrielkay**
### Problem?
#24 

### Why?
To experiment with displaying the posts in different ways on the homepage

### What?
- Added 'Top' button to navbar to sort by top voted posts
- When sorting by new, there's a limit of 5 posts per page, with next and previous buttons to navigate between pages

### CODE REVIEW

File: app/controllers/posts_controller.rb [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
-    @posts = Post.all
+    if params[:sort] == 'top'
+      @posts = Post.all
+      descending = -1
+      @posts = @posts.sort_by do |post|
+        post.post_points * descending
+      end
+    else
+      @page = (params[:page] || 0).to_i
+      @posts = Post.offset(PAGE_SIZE * @page).limit(PAGE_SIZE)
```
Comment (albaer) [inline app/controllers/posts_controller.rb]: Does this only paginate if there isn't any sorting?

File: app/views/posts/index.html.erb [hunk 2]
```diff
@@ -1,3 +1,11 @@
...
   <%= render partial: "post", collection: @posts, as: :cur_post %>
 </div>
+<% if :sort != 'top' %>
+  <div>
+    <span>
+      <%= link_to_if @page > 0, "<< Previous", posts_path(page: @page - 1) if @page != 0 %>
+      <%= link_to "Next >>", posts_path(page: @page + 1) if @posts.size == 5 %>
+    </span>
+  </div>
+<% end %>
```
Comment (albaer) [inline app/views/posts/index.html.erb]: Could you both paginate and sort and avoid having an if statement here?

File: app/controllers/posts_controller.rb [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+  PAGE_SIZE = 5
+
   def index
-    @posts = Post.all
+    if params[:sort] == 'top'
+      @posts = Post.all
+      descending = -1
+      @posts = @posts.sort_by do |post|
+        post.post_points * descending
+      end
```
Comment (albaer) [inline app/controllers/posts_controller.rb]: As we talked about earlier, I think this is a really good start! As part of your next work, a good exercise would be moving the entire sorting mechanism to the database.

File: app/controllers/posts_controller.rb (hunk 1 above)
Comment (gabrielkay) [reply app/controllers/posts_controller.rb]: That's how it is right now. I wasn't sure how to paginate after already having all the posts sorted. I'm looking into it right now

Comment (gabrielkay) [reply app/controllers/posts_controller.rb]: Also I set up both mainly to practice both things, but am looking at combining

Comment (gabrielkay) [reply app/controllers/posts_controller.rb]: Done 🔥

**Pull request merged**